bika/lims/idserver.py

```diff
@@ -189,10 +192,23 @@ def next_id(prefix):
         new_id = next_id(prefix)
         return ('%s' + separator + '%s') % (prefix, new_id)
 
+
 def renameAfterCreation(obj):
+    """
+    Renaming object right after it is created...
+    In some cases we may want to override generate_id function. To do this,
+    an adapter must be added (providing bika.lims.interfaces.IIdServer) for
+    that content type.
+    """
     # Can't rename without a subtransaction commit when using portal_factory
     transaction.savepoint(optimistic=True)
     # The id returned should be normalized already
-    new_id = generateUniqueId(obj)
+    new_id = None
+    # Checking if an adapter exists for this content type. If yes, we will
+    # get new_id from adapter.
+    for name, adapter in getAdapters((obj, ), IIdServer):
+        new_id = adapter.generate_id(obj.portal_type)
```

Comment:
It would be a good idea to add a log if more than one adapters are found.
Something like that should be enough
```
for name, adapter in getAdapters((obj, ), IIdServer):
    if new_id is not None:
        logger.warn('More than one id server adapters found for {} content type.'.format(obj.gettype()))
    new_id = adapter.generate_id(obj.portal_type)

```